library/net_sockets.c

```diff
@@ -535,6 +535,13 @@ int mbedtls_net_recv_timeout( void *ctx, unsigned char *buf, size_t len,
     if( fd < 0 )
         return( MBEDTLS_ERR_NET_INVALID_CONTEXT );
 
+    /* A limitation of select() is that it only works with file descriptors
+     * that are strictly less than FD_SETSIZE. This is a limitation of the
+     * fd_set type. Error out early, because attempting to call FD_SET on a
+     * large file descriptor is a buffer overflow on typical platforms. */
+    if( fd >= FD_SETSIZE )
+        return( MBEDTLS_ERR_NET_POLL_FAILED );
```

Comment:
The CI points out there's no such error code in the 2.7 branch, I think you mean `MBEDTLS_ERR_NET_RECV_FAILED`. But now that my attention has been drawn to it, it's probably incorrect in the other branches as well.

Reply:
I amended the commit to return `MBEDTLS_ERR_NET_RECV_FAILED`. I rewrote the history to avoid having a non-building commit.